osu.Game/Screens/OnlinePlay/Components/SelectionPollingComponent.cs

```diff
@@ -47,9 +48,11 @@ protected override Task Poll()
 
             pollReq.Success += result =>
             {
-                var rooms = new List<Room>(roomManager.Rooms);
+                // existing rooms need to be ordered by their position because the received of NotifyRoomsReceives expects to be able to sort them based on this order.
```

Comment:
```suggestion
                // existing rooms need to be ordered by their position because the receiver of NotifyRoomsReceived expects to be able to sort them based on this order.
```

also I'm having trouble imagining the scenario in which this might be a problem and there's no easy way to test this unless enough lobbies are open - think it might be possible to write a test for this?

Reply:
It's basically 100% a problem: the `BindableList` in `RoomManager` is adding and removing rooms in its own order, disregarding thing incoming order. Every new room will be added at the bottom of the list, but with a `Position` value to track that it's actually at the top. I think this is all a workaround for the fact `BindableList` has no `Insert` function (@smoogipoo should be able to confirm).

The whole flow feels really haphazard so I'd like to see things fixed up at some point (the `ignoredRooms` thing is also really weird). Will see if it's feasible to add a test in the mean time.

Reply:
Attempted to make test coverage for over an hour, but not worth it. You're welcome to have an attempt.

Reply:
The code wasn't originally this way, this is basically an artifact of overengineering and isn't because BindableList had(/has?) no insert method. `RoomManager.onRoomsReceived()` needs a thorough cleanup so that this method doesn't need to return _all_ the rooms.